### PR TITLE
[kjobctl] Add interactive builder

### PR DESCRIPTION
**Pull request by IrvingMg**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add interactive builder for kjobctl.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/experimental/kjobctl/pkg/builder/interactive_builder.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	// TODO: Move to common function to generate a Pod Spec
+	for _, vb := range b.volumeBundles {
+		for _, volume := range vb.Spec.Volumes {
+			index := slices.IndexFunc(pod.Spec.Volumes, func(v corev1.Volume) bool {
+				return v.Name == volume.Name
+			})
+			if index != -1 {
+				pod.Spec.Volumes[index] = volume
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/pkg/builder/interactive_builder.go:67]: What if we have `PodTemplate` with already exists `Volume` that have the same name that we have on `VolumeBundle`?

[the remark this answers is not in the document]
Comment (trasc) [reply cmd/experimental/kjobctl/pkg/builder/interactive_builder.go:67]: We can generate an error or let the API server do it. We should not just silently overwrite, in my opinion.

File: cmd/experimental/kjobctl/pkg/builder/interactive_builder.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+			Labels:       map[string]string{},
+		},
+		Spec: template.Template.Spec,
+	}
+
+	if b.profile != nil {
+		pod.Labels[constants.ProfileLabel] = b.profile.Name
+	}
+
+	// TODO: Move to common function to generate a Pod Spec
```
[the remark this answers is not in the document]
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/builder/interactive_builder.go]: Done

File: cmd/experimental/kjobctl/pkg/builder/interactive_builder.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/builder/interactive_builder.go]: Changed. Option 2 was implemented, concatenating slices and delegating the duplicate name detection to the `validate` command.

File: cmd/experimental/kjobctl/pkg/builder/helpers.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	for _, b := range bundles {
+		volumeBundle.Spec.Volumes = append(volumeBundle.Spec.Volumes, b.Spec.Volumes...)
+		volumeBundle.Spec.ContainerVolumeMounts = append(volumeBundle.Spec.ContainerVolumeMounts, b.Spec.ContainerVolumeMounts...)
+		volumeBundle.Spec.EnvVars = append(volumeBundle.Spec.EnvVars, b.Spec.EnvVars...)
+	}
+
+	return volumeBundle
+}
+
+func buildPodSpec(templateSpec corev1.PodSpec, command []string, requests corev1.ResourceList, bundle v1alpha1.VolumeBundle) corev1.PodSpec {
```
[the remark this answers is not in the document]
Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/builder/helpers.go]: Done

**Review by trasc — COMMENTED** [review]
/lgtm

**Comment by mbobrovskyi** [comment]
lgtm

File: cmd/experimental/kjobctl/pkg/builder/builder.go [hunk 4]
```diff
@@ -215,3 +218,36 @@ func (b *Builder) Do(ctx context.Context) (runtime.Object, error) {
 
 	return bImpl.build(ctx)
 }
+
+func (b *Builder) buildPodSpec(templateSpec corev1.PodSpec) corev1.PodSpec {
+	bundle := mergeBundles(b.volumeBundles)
+
+	templateSpec.Volumes = append(templateSpec.Volumes, bundle.Spec.Volumes...)
+	for i := range templateSpec.Containers {
```
Comment (alculquicondor) [inline cmd/experimental/kjobctl/pkg/builder/builder.go:226]: volume mounts should also be added to initContainers.

But the command should only be added to the first primary container. Requests too.

And this has to be documented somewhere, probably as part of the APIs.

Comment (IrvingMg) [reply cmd/experimental/kjobctl/pkg/builder/builder.go:226]: Done.

File: cmd/experimental/kjobctl/apis/v1alpha1/application_profile_types.go [hunk 5]
```diff
@@ -69,6 +69,8 @@ type SupportedMode struct {
 	// requiredFlags point which cli flags are required to be passed in order to fill the gaps in the templates.
 	// Possible values are cmd, parallelism, completions, request, localqueue.
 	//
+	// cmd and requests values are going to be added only to the first primary container.
```
Comment (mwielgus) [inline cmd/experimental/kjobctl/apis/v1alpha1/application_profile_types.go:72]: The flags should be specific to the used profile. For example, parallelism doesn't make sense for interactive. While it doesn't have to be checked at the CRD level, we probably want to check when running the app.

Comment (trasc) [reply cmd/experimental/kjobctl/apis/v1alpha1/application_profile_types.go:72]: We are checking that the required flags are provided , here: https://github.com/kubernetes-sigs/kueue/blob/398545f971a9d5ecb83d69f410f606525aaa8814/cmd/experimental/kjobctl/pkg/builder/builder.go#L169

**Review by alculquicondor — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**